src/Lock/LockAbstract.php

```diff
@@ -55,12 +55,10 @@ public function __clone()
     public function __destruct()
     {
         foreach ($this->locks as $name => $v) {
-            $released = $this->releaseLock($name);
-            if (!$released) {
-                throw new UnrecoverableMutexException(sprintf(
-                    'Cannot release lock in __destruct(): %s',
-                    $name
-                ));
+            try {
+                $this->releaseLock($name);
+            } catch (Exception $e) {
+                // Ignoring exceptions in destructor to not block shutdown.
             }
```

Comment:
I did a code review on the pull request from @antriver today, and here are what I think:

1. Changes in class _\NinjaMutex\Lock\PredisRedisLock_ do address the edge case in a proper way when locks are not correctly released automatically.

2. When a lock is not released automatically as expected, the _PredisRedisLock_ class can release the lock properly once the lock is expired. Because of that, changes in method _NinjaMutex\Mutex::__destruct()_ and _NinjaMutex\Lock\LockAbstract::__destruct()_ makes sense and make things smooth during execution (especially when handling massive locks). However, ignoring exceptions/errors silently doesn't seem to be a good practice since we lose visibility on them. To me, a better approach is to have a PSR-3 logger optionally attached to the package, log failures like that with the logger without throwing out a new exception. The external logger can be customized to throw out an exception in this case if needed. However, all these are about error reporting and error handling, which is not necessarily to be addressed and this PR.

My suggestion is to adopt changes for class _\NinjaMutex\Lock\PredisRedisLock_, and ignore changes in class _NinjaMutex\Mutex_ and _NinjaMutex\Lock\LockAbstract_ for now. Changes related to error reporting/handling can be addressed later on if needed. Thanks